### ISSUE TITLE
Error in get_report_document: 'decoded_document' is not defined

### ISSUE THREAD

**Issue by olycats** [saleweaver/python-amazon-sp-api#1085]
**Describe the bug**

python-amazon-sp-api==0.18.5

In get_report_document from reports.py, I go error message `'decoded_document' is not defined` here
``` Python
            if download:
                res.payload.update({
                    'document': decoded_document,
                })
            if file:
                self._handle_file(file, decoded_document, character_code)
```

**To Reproduce**

I am not sure if it's possible to reproduce it. See additional context.

**Expected behavior**

Able to download the report without error.

**Desktop (please complete the following information):**

 - OS: Windows 10 using Pycharm and Docker Desktop

**Additional context**

I was impacted by this issue: https://github.com/saleweaver/python-amazon-sp-api/issues/1076
I understand Amazon had resolved the error on their end, so newly requested reports should work without an issue.

However, the way I work with it is to request the reports every day, download the report after it finishes, and retry every hour to download the reports that have not been downloaded. 

So my program is actually retrying to download the report that we requested back on June 21.

I can see that there was a hotfix in 0.18.5, however, I still got the error `'decoded_document' is not defined`.

**Comment by olycats** [comment]
PR: https://github.com/saleweaver/python-amazon-sp-api/pull/1086

**Comment by saleweaver** [comment]
PR is merged and version 0.18.6 is released.
